File: test/lit/ts/arithmetic.ts
// RUN: %cot emit-cir %s | %FileCheck %s

// libtc Phase 1: arithmetic operations

function arith(a: number, b: number): number {
    return a + b;
}

function sub(a: number, b: number): number {
    return a - b;
}

function mul(a: number, b: number): number {
    return a * b;
}

function div(a: number, b: number): number {
    return a / b;
}

function rem(a: number, b: number): number {
    return a % b;
}

// CHECK-LABEL: func.func @arith
// CHECK: cir.add

// CHECK-LABEL: func.func @sub
// CHECK: cir.sub

// CHECK-LABEL: func.func @mul
// CHECK: cir.mul

// CHECK-LABEL: func.func @div
// CHECK: cir.div

// CHECK-LABEL: func.func @rem
// CHECK: cir.rem

File: test/lit/swift/arithmetic.swift
// RUN: %cot emit-cir %s | %FileCheck %s

// libsc: arithmetic operations +, -, *, /, %

func allOps(a: Int32, b: Int32) -> Int32 {
    let sum: Int32 = a + b
    let diff: Int32 = a - b
    let prod: Int32 = sum * diff
    let quot: Int32 = prod / b
    let rem: Int32 = quot % b
    return rem
}

// CHECK-LABEL: func.func @allOps
// CHECK: cir.add %arg0, %arg1 : i32
// CHECK: cir.sub %arg0, %arg1 : i32
// CHECK: cir.mul
// CHECK: cir.div
// CHECK: cir.rem
// CHECK: return
